sql/type.go

```diff
@@ -375,6 +378,24 @@ func (t jsonT) Compare(a interface{}, b interface{}) int {
 	return bytes.Compare(a.([]byte), b.([]byte))
 }
 
+type tupleT struct{}
+
+func (t tupleT) Type() query.Type {
+	return sqltypes.Expression
+}
+
+func (t tupleT) SQL(v interface{}) sqltypes.Value {
+	panic("unable to convert tuple type to SQL")
+}
+
+func (t tupleT) Convert(v interface{}) (interface{}, error) {
+	return nil, fmt.Errorf("unable to convert tuple type")
+}
+
+func (t tupleT) Compare(a, b interface{}) int {
+	panic("unable to compare two tuple values")
```

Comment:
Tuples are comparable, right? `SELECT (3,4) IN ((1,2), (3,5),(3,4)) as b;`

Reply:
We don't have the concrete types to perform the comparison here. It could be done, but some things need to change a bit for that.
- Tuple type would need to be a Type constructor instead, to give it the concrete types of the elements in the tuple.
- `Compare` needs to change to also return an error because there is a lot of cases in which a comparison between tuples can fail (having different number of elements, for example).

Reply:
Do we need comparison now? If not we can just open an issue to add it in the future.

Reply:
It's something that's expected of tuples. If not, they're not very useful. Right now, since we're not using tuples yet, I think this can wait until we do and then do the aforementioned changes, which are vital to implement this correctly.